src/saving/ISaveUpgradeStep.cs

```diff
@@ -162,6 +163,174 @@ private void UpgradeBehaviouralValues(JProperty property, JEnumerable<JProperty>
         }
     }
 
+    internal class UpgradeStep057To058 : BaseRecursiveJSONWalkerStep
+    {
+        protected override string VersionAfter => "0.5.8.0-alpha";
+
+        protected override void CheckAndUpdateProperty(JProperty property)
+        {
+            var children = property.Value.Children<JProperty>();
+            var childrenNames = children.Select(c => c.Name).ToList();
+
+            // Add volume for named patches (properties)
+            if (childrenNames.Contains("Depth") && !childrenNames.Contains("Volume"))
+            {
+                UpgradePatchesVolume((JObject)property.Value, children);
+            }
+
+            // Most patches are defined through references to non-named tokens of array "Adjacent"
+            if (property.Name == "Adjacent")
```

Comment:
I'm a bit suspicious about this as I thought that the base step should already recurse through everything... so in my head, this design should not be required. It seems the call to the base CheckAndUpdateProperty is missing in this method.